### PR TITLE
change datetime format for specific datetime locales

### LINKED ISSUE #6155: Japanese timestamps are in incorrect format on MAPs/Articles/PGLs/STYs
**Describe the bug**
Japanese timestamps are in incorrect format on MAPs (Raised by editorial team)

This affects: 

- Japanese
- Korean
- UK China
- Zhongwen

This affects timestamps on these asset types: 

- CPS MAP: https://www.bbc.com/japanese/video-52227521
- Optimo Article: https://www.bbc.com/japanese/articles/cj4m7n5nrd8o
- CPS PGL: https://www.bbc.com/japanese/features-and-analysis-50914393
- CPS STY: (only on local, not yet audience-facing)
- This is *not* an issue for Story Promo timestamps on Front pages (www.bbc.com/japanese) or on related content on other pagetypes

Information on the source of this issue is in the `Additional context` section at the end of this description.

**To Reproduce**
Steps to reproduce the behaviour:

1. Go to https://www.bbc.com/japanese/video-52227521
2. Scroll down to timestamp
4. See timestamp as:
`9 四月 2020`

**Expected behaviour**
The correct time stamp should be 2020年4月9日 i.e. YYYY年MM月DD日

（年=year, 月=month, 日=day）

**Screenshots**
Screenshot 2020-04-10 at 10 31 01

Media Asset Page timestamp (using ArticleTimestamp component)
`3 十二月 2019`
Screenshot Media Asset Page timestamp

The same bug is found on Optimo articles (using ArticleTimestamp component)
`2 十二月 2019`
Screenshot 2020-04-10 at 10 36 52

But not the front page (using StoryPromo)
https://www.bbc.com/japanese
Screenshot front page

**Testing notes**

- [ ] This bug fix is expected to need manual testing.

**Additional context**
We need to fix the format of the `ArticleTimestamp` component for Japanese, which is used for these pagetypes:

- Optimo Articles
- CPS Media Asset Pages
- CPS Photogalleries
- CPS Stories

Currently, we're using `D MMMM YYYY` for Article Timestamp for all services. See this file: https://github.com/bbc/simorgh/blob/latest/src/app/containers/ArticleTimestamp/timeFormats.js#L5-L8

```
// 22 March 2019
export const formatDate = 'D MMMM YYYY';

// 22 March 2019, 17:05 BST
export const formatDateAndTime = 'D MMMM YYYY, HH:mm z';
````

For Japanese, we should be using:
`formatDate = 'LL';` , which maps to the correct `'YYYY年M月D日'`
`formatDateAndTime = 'LLL';`  which maps to `'YYYY年M月D日 HH:mm'`

Moment file for Japanese: https://github.com/moment/moment/blob/develop/src/locale/ja.js#L17

We should see if we can use LL/LLL for all other locales, too.

### PR DESCRIPTION
**Pull request by jroebu14**
Resolves https://github.com/bbc/simorgh/issues/6155

**Overall change:**
Fix the format of the ArticleTimestamp component for Japanese, Korean, UK China, Zhongwen which is used for these pagetypes:

- Optimo Articles
- CPS Media Asset Pages
- CPS Photogalleries
- CPS Stories

**Code changes:**

- Adds a list of exceptions `'ja', 'ko', 'zh-cn'` of datetime locales that will use LL/LLL as the datetime format
- Fixed tests

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] I have added labels to this PR for the relevant pod(s) affected by these changes
- [x] I have assigned this PR to the Simorgh project

**Testing:**

- [ ] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false npm run test:e2e:interactive`)
- [ ] This PR requires manual testing

### CODE REVIEW

File: src/app/containers/ArticleTimestamp/timeFormats.js [hunk 1]
```diff
@@ -1,8 +1,20 @@
 // 2019-03-22
 export const formatDateNumeric = 'YYYY-MM-DD';
 
+const longDatetimeLocales = ['ja', 'ko', 'zh-cn', 'zh-cn'];
```
Comment (sareh) [inline src/app/containers/ArticleTimestamp/timeFormats.js]:
```suggestion
const longDatetimeLocales = ['ja', 'ko', 'zh-cn'];
```

File: src/app/containers/ArticleTimestamp/timeFormatTests/generateTimeFormats.js [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import fs from 'fs';
 import path from 'path';
 import { timestampsFixtures, format } from './testUtils';
-import services from '#server/utilities/serviceConfigs';
+import services from '../../../../server/utilities/serviceConfigs';
```
Comment (sareh) [inline src/app/containers/ArticleTimestamp/timeFormatTests/generateTimeFormats.js:4]:
```suggestion
import services from '#server/utilities/serviceConfigs';
```

File: src/app/containers/ArticleTimestamp/helpers.js [hunk 3]
```diff
@@ -14,18 +14,22 @@ export const isSameDay = (dayToCompare, timestamp) => {
 
 export const isToday = (timestamp) => isSameDay(Date.now(), timestamp);
 
-export const formatType = ({ lastPublished = null, firstPublished }) => {
+export const formatType = ({
+  lastPublished = null,
+  firstPublished,
+  datetimeLocale,
```
Comment (sareh) [inline src/app/containers/ArticleTimestamp/helpers.js]: Do we want to set a default locale here?

**Review by sareh — COMMENTED** [review]
This all looks good. I've just added some tiny comments.

File: src/app/containers/ArticleTimestamp/timeFormatTests/generateTimeFormats.js (hunk 2 above)
Comment (jroebu14) [reply src/app/containers/ArticleTimestamp/timeFormatTests/generateTimeFormats.js:4]: the babel-node transpilation fails because of the dir alias 😞

Comment (sareh) [reply src/app/containers/ArticleTimestamp/timeFormatTests/generateTimeFormats.js:4]: Oh right, I see.

File: src/app/containers/ArticleTimestamp/helpers.js (hunk 3 above)
Comment (jroebu14) [reply src/app/containers/ArticleTimestamp/helpers.js]: `en-gb`?

**Review by karinathomasbbc — APPROVED** [review]
Looks good 👍

**Review by rebeccamcginn — APPROVED** [review]

**Review by sareh — APPROVED** [review]

**Pull request merged**